nuxeo-core/nuxeo-core-mongodb/src/test/java/org/nuxeo/ecm/core/mongodb/TestMongoDBIndices.java

```diff
@@ -0,0 +1,128 @@
+public class TestMongoDBIndices {
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = coreSession.createDocumentModel("/", "Folder", "Folder");
+        domain = coreSession.createDocument(domain);
+        addPermissions(domain, "james");
+        addPermissions(domain, "bob");
+        coreSession.save();
+    }
+
+    @Test
+    public void shouldFailsWhenCreateExistingChildNameDocument() {
+        try (CloseableCoreSession bobSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(), "bob")) {
+            createDocument(bobSession, domain, DOCUMENT_NAME);
+        }
+
+        try (CloseableCoreSession jamesSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(),
+                "james")) {
+            createDocument(jamesSession, domain, DOCUMENT_NAME);
+            fail("should throw a ConcurrentUpdateException");
+        } catch (ConcurrentUpdateException cue) {
+            assertEquals(SC_CONFLICT, cue.getStatusCode());
+            assertTrue(
+                    cue.getMessage()
+                       .contains(String.format(
+                               "E11000 duplicate key error collection: %s.test index: ecm:parentId_1_ecm:name_1 dup key: { : \"%s\", : \"%s\" }",
+                               DEFAULT_MONGODB_DBNAME, domain.getId(), DOCUMENT_NAME)));
+        }
+    }
+
+    @Test
+    public void shouldFailsWhenMovingDocumentToExistingChildName() {
+        try (CloseableCoreSession bobSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(), "bob")) {
+            createDocument(bobSession, domain, DOCUMENT_NAME);
+        }
+
+        try (CloseableCoreSession jamesSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(),
+                "james")) {
+            DocumentModel documentModel = createDocument(jamesSession, domain, "jamesFileName");
+
+            // org.nuxeo.ecm.core.storage.dbs.DBSSession.move (Objects.equals(oldParentId, parentId)) checks the unique
+            // name before calling the mongodb update backend
+            documentModel = jamesSession.move(documentModel.getRef(), null, DOCUMENT_NAME);
+            jamesSession.saveDocument(documentModel);
+            jamesSession.save();
+            fail("should throw a DocumentExistsException");
+        } catch (DocumentExistsException dee) {
+            assertEquals(String.format("Destination name already exists: %s", DOCUMENT_NAME), dee.getMessage());
+        }
+    }
+
+    protected DocumentModel createDocument(CoreSession session, DocumentModel parent, String fileName) {
+        DocumentModel doc = session.createDocumentModel(parent.getPathAsString(), fileName, "File");
+        session.createDocument(doc);
+        session.save();
```

Comment:
Not needed

Reply:
you mean the whole method, if it's the case i created it just to be more concise about the test and keep the creation process outer and to avoid duplicate of some code.

Reply:
No, the line. You're doing a save at the end of `@Before` method, this one should be enough, if not due to ACLs, put save before adding ACLs.

Furthermore, AFAIK it's better to get the document from createDocument as it may contain data that the input won't have.

Reply:
i test with removing the `session.save()`  on `createDocument ` and my test failed because the second creation will be able to put the same name as our session is not saved. for your other suggestion of adding `save` before `addPermission` i did it